sdk/daml-script/test/daml/upgrades/DataChanges.daml

```diff
@@ -77,7 +77,7 @@ templateInvalidChange shouldSucceed makeV1Contract v2Choice = test $ do
   case (res, shouldSucceed) of
     (Right "V2", True) -> pure ()
     (Left (WronglyTypedContract {}), False) -> pure ()
-    -- IDE Ledger doesn't apply obfuscation, instead the lookup error is wrapped in SCrash
+    (Left (UpgradeError (DowngradeFailed _) msg), False) -> pure ()
     (Left (UnknownError msg), False) | "SErrorCrash" `isInfixOf` msg -> pure ()
```

Comment:
I'm unclear why I need to have this particular case? Without it, IDE Ledger testing will fail.

Reply:
You probably just need to add a case to the IDE Ledger Error translation. I think Canton unwraps the error and makes it nice, but IDE doesn't.